Muart-C/politico
# Ch add parties edge cases 163884011

**What does this PR do?**
 - ensures that add party endpoint accepts sanitized data

**Description of Task to be completed?**
- A party is created by sending a **POST** request to the endpoint _127.0.0.1:500/api/v1/parties_ with the payload below;
  ``` 
        {
	    "name" : "nasa",
	    "logoUrl" : "https://gcom.com",
	    "hqAddress" : "Machakos" 
        }
  ```

- A response with a payload of the form below is returned if successful

```
 {
  "data": [
    {
      "hq_address": "Machakos",
      "id": 2,
      "logo_url": "https://gcom.com",
      "name": "nasa"
    }
  ],
  "message": "party nasa was created",
  "status": 201
}
```
- A response with a payload of the form below is returned if the request is not successful 

   ```
      {
            “status” : 400 ,
            “error” : "respective error message based on the problem”
      }
   ```

**How should the app be tested?**

-  `git clone https://github.com/Muart-C/politico.git`

- `cd politico`

- `git checkout ch-add-parties-edge-cases-163884011`

- run `pytest --cov=api tests/test_party.py`
- All the tests should pass



**What are the relevant pivotal tracker stories***
https://www.pivotaltracker.com/story/show/163884011

**screenshots**

![screenshot from 2019-02-12 10-39-27](https://user-images.githubusercontent.com/28337458/52619359-801dbe00-2eb2-11e9-973f-9ae9df4d5a88.png)
